content/300-guides/050-database/850-using-prisma-with-planetscale.mdx

```diff
@@ -0,0 +1,240 @@
+
+Note that you can only push new changes to a development branch, not a production branch. If you try to push to a production branch, you will get the [error message](/reference/api-reference/error-reference#p3022) `Direct execution of DDL (Data Definition Language) SQL statements is disabled on this database.` Once you are happy with the changes you make on a development branch, you can create a deploy request, which you can then review before deploying the changes to production.
+
+## How to create indexes on foreign keys
+
+When [emulating referential integrity in Prisma](#how-to-enable-emulation-of-referential-integrity), you will need to create your own indexes. As an example of a situation where you would want to an add an index, take this schema for a blog with posts and comments:
+
+```prisma file=schema.prisma
+model Post {
+  id       Int       @id @default(autoincrement())
+  title    String
+  content  String
+  likes    Int       @default(0)
+  comments Comment[]
+}
+
+model Comment {
+  id      Int    @id @default(autoincrement())
+  comment String
+  postId  Int
+  post    Post   @relation(fields: [postId], references: [id], onDelete: Cascade)
+}
+```
+
+The `postId` field in the `User` model refers to the corresponding `id` field in the `Post` model. However this is not implemented as a foreign key in PlanetScale, so it doesn't have an index. This means that some queries may not be well optimised. For example, if you query for all comments with a certain post `id`, PlanetScale may have to do a full table lookup. This could be slow, and also expensive because PlanetScale's billing model charges for the number of rows read.
+
+To avoid this, you can define an index on the `postId` field using [Prisma's `@@index` argument](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#index):
+
+```prisma file=schema.prisma highlight=15;add
+model Post {
+  id       Int       @id @default(autoincrement())
+  title    String
+  content  String
+  likes    Int       @default(0)
+  comments Comment[]
+}
+
+model Comment {
+  id      Int    @id @default(autoincrement())
+  comment String
+  postId  Int
+  post    Post   @relation(fields: [postId], references: [id], onDelete: Cascade)
+
+  @@index([postId])
+}
+```
+
+You can then add this change to your schema [using `db push`](#how-to-make-schema-changes-with-db-push).
+
+One issue to be aware of is that [implicit many-to-many relations](/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) cannot have an index added in this way. If query speed or cost is an issue, you may instead want to use an [explicit many-to-many relation](/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations) in this case.
```

Comment:
This could be a Note or Admonition or Warning or something like that.

Reply:
Done